### PR TITLE
Support building locally

### PR DESCRIPTION
**Pull request by andreim-sv**
### Purpose
Enable local build
### Detail
Added a few touches to allow for local build:
  - NLohman's Json can be in a  custom location;
  - 3rd party Prometheus headers are now wrapped in enable/disable `effc++` set of compiler warnings

The latter allowed to re-enable the warnings for our code.

### CODE REVIEW

File: src/comm/ConnClient.cc [hunk 1]
```diff
@@ -49,7 +49,9 @@ using namespace comm;
 
 ConnClient::ConnClient(const Address& server_address, ConnClientConfig config) :
     _config(std::move(config)),
-    _server(std::move(server_address))
+    _connection(),
+    _server(std::move(server_address)),
+    _ssl_ctx()
```
Comment (luisremis) [inline src/comm/ConnClient.cc:54]:
```suggestion
    _ssl_ctx(create_client_context())
```

File: src/comm/ConnClient.cc [hunk 2]
```diff
@@ -49,7 +49,9 @@ using namespace comm;
...
 {
     _ssl_ctx = create_client_context();
```
Comment (luisremis) [inline src/comm/ConnClient.cc:56]:
```suggestion
```

File: src/comm/ConnServer.cc [hunk 3]
```diff
@@ -51,8 +51,10 @@ using namespace comm;
 
 ConnServer::ConnServer(int port, ConnServerConfig config) :
     _config(std::move(config)),
+    _listening_socket(),
     _open_ssl_initializer(OpenSSLInitializer::instance()),
-    _port(port)
+    _port(port),
+    _ssl_ctx()
```
Comment (luisremis) [inline src/comm/ConnServer.cc:57]:
```suggestion
    _ssl_ctx(create_server_context())
```

File: src/comm/ConnServer.cc [hunk 4]
```diff
@@ -51,8 +51,10 @@ using namespace comm;
...
 {
     _ssl_ctx = create_server_context();
```
Comment (luisremis) [inline src/comm/ConnServer.cc:59]:
```suggestion
```

**Review by luisremis — APPROVED** [review]
looks good, minor suggestion.

File: src/comm/ConnClient.cc (hunk 2 above)
Comment (andreim-sv) [reply src/comm/ConnClient.cc:56]: Done.

File: src/comm/ConnServer.cc (hunk 3 above)
Comment (andreim-sv) [reply src/comm/ConnServer.cc:57]: Done

File: src/comm/ConnServer.cc (hunk 4 above)
Comment (andreim-sv) [reply src/comm/ConnServer.cc:59]: Done

**Pull request merged**